app/uk/gov/hmrc/cgtpropertydisposals/controllers/SubscriptionController.scala

```diff
@@ -52,24 +56,30 @@ class SubscriptionController @Inject()(service: SubscriptionService, cc: Control
                                       RequestValidationError(s"Could not parse JSON body as subscription request: $e")
                                   )
                               )
-        subscriptionResponse <- service.subscribe(subscriptionDetails).leftMap[SubscriptionError](BackendError(_))
+        subscriptionResponse <- subscriptionService
+                                 .subscribe(subscriptionDetails)
+                                 .leftMap[SubscriptionError](BackendError(_))
+        _ <- taxEnrolmentService
+              .allocateEnrolmentToGroup(subscriptionResponse.cgtReferenceNumber, subscriptionDetails)
+              .leftMap[SubscriptionError](_ => EnrolmentError(subscriptionResponse.cgtReferenceNumber))
       } yield subscriptionResponse
 
     result.fold(
       {
         case RequestValidationError(msg) =>
           logger.warn(s"Error in request to subscribe: $msg")
           BadRequest
-
         case BackendError(e) =>
           logger.warn("Error while trying to subscribe", e)
           InternalServerError
+        case EnrolmentError(cgtReference) =>
+          logger.warn("Error while trying to allocate enrolment")
+          Ok(Json.toJson(cgtReference)) // We return a 200 OK even though it is an error as we will retry asynchronously
```

Comment:
the only thing missing now is a unit test for this scenario

Reply:
added test